bio_hansel/scripts/main.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import logging
+import os
+import sys
+
+from extract_test_columns import extract_test_columns
+from find_cluster import find_clusters
+from fisher_test import fisher_test
+from get_sequence import get_sequences
+from read_vcf import read_vcf
+from split_genomes import split_genomes
+
+SCRIPT_NAME = 'schema_creation'
+LOG_FORMAT = '%(asctime)s %(levelname)s: %(message)s [in %(pathname)s:%(lineno)d]'
+
+
+def init_console_logger(logging_verbosity=3):
+    logging_levels = [logging.ERROR, logging.WARN, logging.INFO, logging.DEBUG]
+    if logging_verbosity > (len(logging_levels) - 1):
+        logging_verbosity = 3
+    lvl = logging_levels[logging_verbosity]
+
+    logging.basicConfig(format=LOG_FORMAT, level=lvl)
+
+
+def init_parser():
+    parser = argparse.ArgumentParser(
+        prog=SCRIPT_NAME, formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument(
+        '-i',
+        '--input-vcf',
+        required=True,
+        help=
+        'vcf file containing the list of SNVs that are related to the genomes')
+
+    parser.add_argument(
+        '-g',
+        '--reference-genome-file',
+        required=True,
+        help='Path to reference genome name, can be .gb or .gbk format')
+    parser.add_argument(
+        '-o',
+        '--output-folder-name',
+        required=True,
+        help=
+        'Output file name for the genomes, default folder name is /usr/home/genomes'
+    )
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action='count',
+        default=0,
+        help=
+        'Logging verbosity level (-v == show warnings; -vvv == show debug info)'
+    )
+
+    parser.add_argument(
+        '-n',
+        '--schema-name',
+        help=
+        'A unique name for the schema file that is generated, the default is just'
+        ' {bio_hansel-schema-reference_genome_name}-{schema_version}'
+    )
+
+    parser.add_argument(
+        '-m',
+        '--schema-version',
+        help=
+        'An optional version number for the schema file that is generated'
+
+    )
+
+    parser.add_argument(
+        '-t',
+        '--minimum-threshold',
+        required=True,
+        type=float,
+        help=
+        'Minimum threshold to be tested using the hierarchical clustering scheme'
+    )
+
+    parser.add_argument(
+        '-u',
+        '--maximum-threshold',
+        required=True,
+        type=float,
+        help=
+        'Maximum threshold to be tested using the hierarchical clustering scheme'
+    )
+
+    return parser
+
+
+def main():
+    home_folder = os.path.expanduser('~')
```

Comment:
Please remove this line. We are not interested in the user's home folder. The user can specify the output directory destination via the command-line as an option (`--outdir whatever-dir-I-want`).

Reply:
fixed addressed in commit 4277dffbf5dc652244a1425057a3cee485b4b4e1